src/main/java/it/eng/idsa/businesslogic/configuration/WebSocketServerConfiguration.java

```diff
@@ -1,70 +1,80 @@
 package it.eng.idsa.businesslogic.configuration;
 
+import it.eng.idsa.businesslogic.processor.consumer.websocket.server.*;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Scope;
-import org.springframework.context.annotation.ScopedProxyMode;
-
-import it.eng.idsa.businesslogic.processor.consumer.websocket.server.FileRecreatorBeanServer;
-import it.eng.idsa.businesslogic.processor.consumer.websocket.server.FrameBufferBean;
-import it.eng.idsa.businesslogic.processor.consumer.websocket.server.IdscpServerBean;
-import it.eng.idsa.businesslogic.processor.consumer.websocket.server.InputStreamSocketListenerServer;
-import it.eng.idsa.businesslogic.processor.consumer.websocket.server.RecreatedMultipartMessageBean;
-import it.eng.idsa.businesslogic.processor.consumer.websocket.server.ResponseMessageBufferBean;
-import it.eng.idsa.businesslogic.processor.consumer.websocket.server.ResponseMessageSendPartialServer;
-import it.eng.idsa.businesslogic.processor.producer.websocket.client.FileStreamingBean;
-import it.eng.idsa.businesslogic.processor.producer.websocket.client.IdscpClientBean;
-import it.eng.idsa.businesslogic.processor.producer.websocket.client.InputStreamSocketListenerClient;
 
 /**
- * 
  * @author Milan Karajovic and Gabriele De Luca
- *
  */
 
 @Configuration
 public class WebSocketServerConfiguration {
-	
-	@Bean
-	@Scope("singleton")
-	public FrameBufferBean frameBufferWebSocket() {
-		return new FrameBufferBean();
-	}
-	
-	@Bean
-	@Scope("singleton")
-	public InputStreamSocketListenerServer inputStreamSocketListenerWebSocketServer() {
-		return new InputStreamSocketListenerServer();
-	}
-	
-	@Bean
-	@Scope("singleton")
-	public IdscpServerBean idscpServerWebSocket() {
-		return new IdscpServerBean();
-	}
-	
-	@Bean
-	@Scope("singleton")
-	public FileRecreatorBeanServer fileRecreatorBeanWebSocket() {
-		return new FileRecreatorBeanServer();
-	}
 
-	@Bean
-	@Scope("singleton")
-	public RecreatedMultipartMessageBean recreatedMultipartMessageBeanWebSocket() {
-		return new RecreatedMultipartMessageBean();
-	}
-	
-	@Bean
-	@Scope("singleton")
-	public ResponseMessageBufferBean responseMessageBufferWebSocket() {
-		return new ResponseMessageBufferBean();
-	}
+    private FrameBufferBean frameBufferBean;
+
+    @Bean
+    @Scope("singleton")
+    public FrameBufferBean frameBufferWebSocket() {
+       return new FrameBufferBean();
+    }
+
+    @Bean
+    @Scope("singleton")
+    public InputStreamSocketListenerServer inputStreamSocketListenerWebSocketServer() {
+        return new InputStreamSocketListenerServer();
+    }
+
+    @Bean
+    @Scope("singleton")
+    public IdscpServerBean idscpServerWebSocket() {
+        return new IdscpServerBean();
+    }
+
+    /**
+     * @author Antonio Scatoloni
+     * @return
+     */
+    @Bean
+    @Scope("singleton")
+    public HttpWebSocketServerBean httpsServerWebSocket() {
```

Comment:
- Does the name of the type shoud be Https ... insted of the Http .... ? The same question in the others parts of the code?
- Maybe, it will be good if we create separated Configuration.java class for the Https beans?

Reply:
It may be Https, but all the solution may work over Http, using SSL is a choice.